nbs/99_manuscript/coefs_comp/py/08_01-gtex_whole_blood-intersections.py

```diff
@@ -0,0 +1,376 @@
+
+# %%
+df_r_data = df_plot
+
+# %%
+df_r_data.shape
+
+# %%
+gene_pairs_by_cats = from_indicators(categories, data=df_r_data)
+
+# %%
+gene_pairs_by_cats
+
+# %%
+fig = plt.figure(figsize=(18, 5))
+
+g = plot(
+    gene_pairs_by_cats,
+    show_counts=True,
+    sort_categories_by=None,
+    element_size=None,
+    fig=fig,
+)
+
+# %% [markdown]
+# ## Sort by categories of subsets
+
+# %%
+df_r_data = df_plot
+
+# %%
+df_r_data.shape
+
+# %%
+gene_pairs_by_cats = from_indicators(categories, data=df_r_data)
+
+# %%
+gene_pairs_by_cats
+
+# %%
+gene_pairs_by_cats = gene_pairs_by_cats.sort_index()
+
+# %%
+_tmp_index = gene_pairs_by_cats.index.unique().to_frame(False)
+display(_tmp_index)
+
+# %%
+_tmp_index[_tmp_index.sum(axis=1) == 3]
+
+# %%
+_tmp_index.apply(lambda x: x[0:3].sum() == 0, axis=1)
+
+# %%
+# agreements on top
+_tmp_index.loc[
+    _tmp_index[
+        _tmp_index.apply(lambda x: x.sum() > 1, axis=1)
+        & _tmp_index.apply(lambda x: x[0:3].sum() == 0, axis=1)
+        & _tmp_index.apply(lambda x: 3 > x[3:].sum() > 1, axis=1)
+    ].index
+].apply(tuple, axis=1).to_numpy()
+
+# %%
+# agreements on bottom
+_tmp_index.loc[
+    _tmp_index[
+        _tmp_index.apply(lambda x: x.sum() > 1, axis=1)
+        & _tmp_index.apply(lambda x: 3 > x[0:3].sum() > 1, axis=1)
+        & _tmp_index.apply(lambda x: x[3:].sum() == 0, axis=1)
+    ].index
+].apply(tuple, axis=1).to_numpy()
+
+# %%
+# diagreements
+_tmp_index.loc[
+    _tmp_index[
+        _tmp_index.apply(lambda x: x.sum() > 1, axis=1)
+        & _tmp_index.apply(lambda x: x[0:3].sum() > 0, axis=1)
+        & _tmp_index.apply(lambda x: x[3:].sum() > 0, axis=1)
+    ].index
+].apply(tuple, axis=1).to_numpy()
+
+# %%
+# order subsets
+gene_pairs_by_cats = gene_pairs_by_cats.loc[
+    [
+        # pairs not included in categories:
+        # (False, False, False, False, False, False),
+        # full agreements on high:
+        (False, False, False, True, True, True),
+        # agreements on top
+        (False, False, False, False, True, True),
+        (False, False, False, True, False, True),
+        (False, False, False, True, True, False),
+        # agreements on bottom
+        (False, True, True, False, False, False),
+        (True, False, True, False, False, False),
+        (True, True, False, False, False, False),
+        # full agreements on low:
+        (True, True, True, False, False, False),
+        # diagreements
+        #   clustermatch
+        (False, True, False, True, False, True),
+        (False, True, False, False, False, True),
+        (True, False, False, False, False, True),
+        (True, True, False, False, False, True),
+        #   pearson
+        (False, False, True, False, True, False),
+        (True, False, False, False, True, False),
+        (True, False, True, False, True, False),
+        #   spearman
+        (False, True, False, True, False, False),
+    ]
+]
+
+# %%
+fig = plt.figure(figsize=(14, 5))
+
+# g = plot(
+g = MyUpSet(
```

Comment:
Interesting! Looks like there is pretty good consistency of the gene ranks across methods which is good or gene pairs are at least consistent across two of the methods. I'm curious about the cases on the right side where there is a switch from high-low between correlation metrics, seems like this is rare but I'm just curious. Are these the ones you plotted in `08_05-gtex_whole_blood-intersections_plots.ipynb`?

Reply:
Take a look at the figure I pasted above. Those relationships in the right are the "disagreements", and on the left the "agreements". And yes, those are the ones I plotted in that notebook!